src/Database/Database.php

```diff
@@ -849,29 +849,11 @@ public function exists(string $table, array $condition): bool
 			return false;
 		}
 
-		$fields = [];
-
 		if (empty($condition)) {
 			return DBStructure::existsTable($table);
 		}
 
-		reset($condition);
-		$first_key = key($condition);
-		if (!is_int($first_key)) {
-			$fields = [$first_key];
-		}
-
-		$stmt = $this->select($table, $fields, $condition, ['limit' => 1]);
-
-		if (is_bool($stmt)) {
-			$retval = $stmt;
-		} else {
-			$retval = ($this->numRows($stmt) > 0);
-		}
-
-		$this->close($stmt);
-
-		return $retval;
+		return $this->count($table, $condition) > 0;
```

Comment:
A `count` is significant slower than just having a look if any database entry exists.

Reply:
There's no "just having a look if any database entry exists" function in MySQL/MariaDB, so either way (Statement row number or COUNT) can be defended. As I understand it, `LIMIT` is mainly useful to reduce the amount of data being transferred from the database to the client, but it can slow down queries.

I prefer the COUNT version as it is simpler both in code and in concept, you're asking the database itself the number of records in the result set produced by the provided condition.

Reply:
`count` will be deadly for large tables, since the system has to scan through the whole database to find each entry. The query with the limit stops at the first occurrence, which will be much faster. See this example:
```
]> select count(*) from `post-user-view` where body like "%Test%";
+----------+
| count(*) |
+----------+
|   265487 |
+----------+
1 row in set (27 min 48,265 sec)

> select id from `post-user-view` where body like "%Test%" limit 1;
+----------+
| id       |
+----------+
| 20406627 |
+----------+
1 row in set (0,004 sec)
```

Reply:
Oof, can’t argue with these numbers. I’ll revert my changes.

Reply:
Done.